runtime/near-evm-runner/src/near_ext.rs

```diff
@@ -59,11 +63,18 @@ impl<'a> NearExt<'a> {
             static_flag,
             depth,
             evm_gas_config,
+            chain_id,
         }
     }
 }
 
 impl<'a> vm::Ext for NearExt<'a> {
+    /// EIP-1344: Returns the current chain's EIP-155 unique identifier.
+    /// See https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1344.md
+    fn chain_id(&self) -> u64 {
```

Comment:
Should be okay, but why previous chain_ids are u128 and here is u64?

Reply:
The `chain_id(&self) -> u64` is OpenEthereum's API that we are implementing here. So, that's what we have to give OpenEthereum. Internally in nearcore, we pass the chain ID around as a `u128`, so I deferred the truncation to the very last step here when we have to hand the value over to OpenEthereum.

For reference, [EIP-1344](https://eips.ethereum.org/EIPS/eip-1344) says _"Chain ID is a 256-bit value"_. However, [all known chain IDs](https://chainid.network) do fit into a `u64`, which may be why the OpenEthereum devs thought that was enough.